### PR TITLE
set parser error when infer type is null

### LINKED ISSUE #38582: No error when trying to assign a non null to variable type hinted null.
**Godot version:**
3.2.1.stable.offical

**OS/device including version:**
win 10 home 10.0.18362 build 18362
HP laptop 15-da0xxx

**Issue description:**
Type hinting a variable to null doesn't give an error when you try to set it to a non null value it just stays null.
**Steps to reproduce:**
Type hint a variable to null with ":=null"
**Minimal reproduction project:**
[Map Editor.zip](https://github.com/godotengine/godot/files/4602411/Map.Editor.zip)

### PR DESCRIPTION
**Pull request by ThakeeNathees**
Fix: #38582
<s>any suggestion for better error message?</s>
**EDIT** error message is: `The variable type cannot be inferred because its value is "null"`
infer-type-null

### CODE REVIEW

**Comment by aaronfranke** [comment]
Would make more sense as "Cannot infer the type of 'null' as it has no type" or similar. [Also...](https://www.youtube.com/watch?v=QEzhxP-pdos)

**Comment by vnen** [comment]
The message can be something like: `The variable type cannot be inferred because its value is "null"`.

**Comment by ThakeeNathees** [comment]
@vnen I've changed it.

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -8211,6 +8215,10 @@ void GDScriptParser::_check_block_types(BlockNode *p_block) {
 							_set_error("The assigned value doesn't have a set type; the variable type can't be inferred.", lv->line);
 							return;
 						}
+						if (assign_type.kind == DataType::BUILTIN && assign_type.builtin_type == Variant::NIL) {
+							_set_error("Invalid assigned value type (\"null\") is not allowed.", lv->line);
```
Comment (vnen) [inline modules/gdscript/gdscript_parser.cpp:8219]: What about this one?

Comment (ThakeeNathees) [reply modules/gdscript/gdscript_parser.cpp]: @vnen sorry didn't notice, and also changed it now.

**Review by vnen — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**